### ISSUE TITLE
Update default to Nextcloud 20

### ISSUE THREAD

**Issue by madonius** [NixOS/nixpkgs#108419]
In line [375](https://github.com/NixOS/nixpkgs/blob/c5c6009fb436efe5732e07cd0e5692f495321752/nixos/modules/services/web-apps/nextcloud.nix#L375) of the nextcloud nixos module the current default is nextcloud 19.
On the one hand, this is not represented in the documentation. On the other, this yields the error that there are unpatched security issues.

**Comment by Ma27** [comment]
Retroactively changing the default on 20.09 is not just a breaking change, but effectively breaks the upgrade path from 20.03: the problem with Nextcloud (and the main reason for this complex package-selection thing) is that they don't support upgrades across multiple major releases (i.e. you can't go from v18 to v20).

The whole point of this `.package` option is that one can install an arbitrary Nextcloud version while we (as maintainers) can set "conservative" defaults for working upgrade paths. In your case it would be sufficient to set `services.nextcloud.package = pkgs.nextcloud20;`.

Regarding CVE-2020-8259 and CVE-2020-8152: both vulnerabilities affect the encryption module which is turned off by default (IIRC it can only be enabled in the admin UI anyways). It's unfortunate that Nextcloud upstream doesn't care about older yet still active majors and rates security issues as "Low" even though the CVEs are rated as "High" by e.g. NIST, but that's where we are.
In other words: if you don't use the affected features it's still OK to use `nextcloud19` (which can be allowed by setting `      nixpkgs.config.permittedInsecurePackages = ["nextcloud-19.0.6"];`.

**Comment by madonius** [comment]
I see your point, thanks.
The issue is that I, as a user, install Nextcloud and see a security warning. Given the above, maybe the message could state that you may use the `.package` option and use nextcloud20. Would that be feasible? Or given the fact that 21.03 is 'around the corner' would it be worth it?

**Comment by Ma27** [comment]
@veprbl why did you mark this as security-relevant? Both CVEs in question are disclosed by upstream and `nixpkgs` will warn about them by default. This is merely an issue with documentation and helpful warnings IMHO.

**Comment by veprbl** [comment]
@Ma27 The issue seems to be about undocumented path towards using nextcloud while staying secure

**Comment by Ma27** [comment]
Yes, it's about documentation, no immediate action is pending to fix a security issue! Nextcloud versions with known vulnerabilities yield an evaluation error, this is primarily about suggesting what to do if one stumbles upon this.
